app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala

```diff
@@ -16,8 +16,71 @@
 
 package uk.gov.hmrc.helptosave.connectors
 
-import uk.gov.hmrc.http.HttpResponse
+import cats.Show
+import cats.syntax.show._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json.{JsNull, JsValue, Writes}
+import uk.gov.hmrc.helptosave.config.{AppConfig, WSHttp}
+import uk.gov.hmrc.helptosave.metrics.Metrics
+import uk.gov.hmrc.helptosave.models.UCResponse
+import uk.gov.hmrc.helptosave.util.Time.nanosToPrettyString
+import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, Logging, NINO, PagerDutyAlerting}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[DESConnectorImpl])
 trait DESConnector {
   def desCorrelationId(response: HttpResponse): String = response.header("CorrelationId").getOrElse("-")
+
+  def isEligible(nino: String, ucResponse: Option[UCResponse])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def setFlag(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def getPersonalDetails(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def getThreshold()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+}
+
+@Singleton
+class DESConnectorImpl @Inject() (http:              WSHttp,
+                                  metrics:           Metrics,
+                                  pagerDutyAlerting: PagerDutyAlerting)(implicit transformer: LogMessageTransformer, appConfig: AppConfig)
+  extends DESConnector with Logging {
+
+  val itmpBaseURL: String = appConfig.baseUrl("itmp-eligibility-check")
+  val itmpEnrolmentURL: String = appConfig.baseUrl("itmp-enrolment")
+  val payeURL: String = appConfig.baseUrl("paye-personal-details")
+  val itmpThresholdURL: String = s"${appConfig.baseUrl("itmp-threshold")}/universal-credits/threshold-amount"
```

Comment:
all these base urls should be pointing to  the the same DES URL , so you can cleanup this

Reply:
good call, thanks, this has now been cleaned up